test/Microsoft.ML.Functional.Tests/Datasets/AllTypes.cs

```diff
@@ -0,0 +1,148 @@
+                        new TextLoader.Column("U1", DataKind.U1, 7),
+                        new TextLoader.Column("I2", DataKind.I2, 8),
+                        new TextLoader.Column("U2", DataKind.U2, 9),
+                        new TextLoader.Column("I4", DataKind.I4, 10),
+                        new TextLoader.Column("U4", DataKind.U4, 11),
+                        new TextLoader.Column("I8", DataKind.I8, 12),
+                        new TextLoader.Column("U8", DataKind.U8, 13),
+                        new TextLoader.Column("R4", DataKind.R4, 14),
+                        new TextLoader.Column("R8", DataKind.R8, 15),
+                        new TextLoader.Column("Tx", DataKind.TX, 16),
+                        new TextLoader.Column("Ts", DataKind.TS, 17),
+                        new TextLoader.Column("Dt", DataKind.DT, 18),
+                        new TextLoader.Column("Dz", DataKind.DZ, 19),
+                        new TextLoader.Column("Ug", DataKind.UG, 20),
+                    },
+                    hasHeader: true,
+                    separatorChar: separator);
+        }
+
+        /// <summary>
+        /// Generate an IEnumerable of AllTypes.
+        /// </summary>
+        /// <param name="numExamples">The number of AllTypesDataset objects to make.</param>
+        /// <param name="seed">The random seed.</param>
+        /// <returns>An IEnumerable of AllTypes.</returns>
+        public static IEnumerable<AllTypes> GenerateDataset(int numExamples = 5, int seed = 1)
+        {
+            var rng = new Random(seed);
+            for (int i = 0; i < numExamples; i++)
+            {
+                yield return new AllTypes
+                {
+                    Label = rng.NextDouble() > 0.5,
+                    Features = new float[] {
+                        (float)rng.NextDouble(),
+                        (float)rng.NextDouble(),
+                        (float)rng.NextDouble(),
+                        (float)rng.NextDouble(),
+                        (float)rng.NextDouble()
+                    },
+                    I1 = (sbyte)rng.Next(),
+                    U1 = (byte)rng.Next(),
+                    I2 = (short)rng.Next(),
+                    U2 = (ushort)rng.Next(),
+                    I4 = rng.Next(),
+                    U4 = (uint)rng.Next(),
+                    I8 = (long)rng.Next(),
+                    U8 = (ulong)rng.Next(),
+                    R4 = (float)rng.NextDouble(),
+                    R8 = (double)rng.NextDouble(),
+                    Tx = GetRandomCharSpan(rng),
+                    Ts = TimeSpan.FromSeconds(rng.NextDouble() * (1 + rng.Next())),
+                    Dt = DateTime.FromOADate(rng.Next(657435, 2958465)),
+                    Dz = DateTimeOffset.FromUnixTimeSeconds((long)(rng.NextDouble() * (1 + rng.Next()))),
+                    Ug = new RowId((ulong)rng.Next(), (ulong)rng.Next())
+                };
```

Comment:
Factor-out into a method - GetRandomInstance(Random rnd, int featuresLength)

Reply:
Factored out, made public with XML tags. Cannot add a featuresLength at this point (see above comment).

---
In reply to: [256542015](https://github.com/dotnet/machinelearning/pull/2518#discussion_r256542015) [](ancestors = 256542015)